src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2909,6 +2910,137 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+
+            ' Built in enum binary operators
+            If leftType.IsEnumType() AndAlso
+               leftType.Equals(rightType, TypeCompareKind.ConsiderEverything) AndAlso
+               leftType.Equals(returnType, TypeCompareKind.ConsiderEverything) Then
+                If opInfo.BinaryOperatorKind = BinaryOperatorKind.Xor OrElse
+                   opInfo.BinaryOperatorKind = BinaryOperatorKind.And OrElse
+                   opInfo.BinaryOperatorKind = BinaryOperatorKind.Or Then
+                    Return
+                End If
+            End If
+
+            ' Quick table access to determine if these types are legal.
+            Dim resolved = OverloadResolution.ResolveNotLiftedIntrinsicBinaryOperator(opInfo.BinaryOperatorKind, leftType.SpecialType, rightType.SpecialType)
+            If resolved <> SpecialType.None Then
+                ' Quick access table strangely maps `string Like string` to the `string` return type. remap it to 'bool'
+                ' here as that's what the operator actually is.
+                '
+                ' Similarly, the relations table doesn't include useful info.  it always has the original type,
+                ' not the expected 'bool' return type.
+                If resolved <> SpecialType.System_Object Then
+                    If opInfo.BinaryOperatorKind = BinaryOperatorKind.Equals OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.NotEquals OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.LessThanOrEqual OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.GreaterThanOrEqual OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.LessThan OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.GreaterThan OrElse
+                       opInfo.BinaryOperatorKind = BinaryOperatorKind.Like Then
+
+                        resolved = SpecialType.System_Boolean
+                    End If
+                End If
+
+                If returnType.SpecialType = resolved Then
+                    Return
+                End If
+            End If
+
+            Throw New ArgumentException($"Unsupported built-in binary operator: {returnType.ToDisplayString()} operator {name}({leftType.ToDisplayString()}, {rightType.ToDisplayString()})")
+        End Sub
+
+        Protected Overrides Function CommonCreateBuiltinOperator(
+                name As String,
+                returnType As ITypeSymbol,
+                operandType As ITypeSymbol) As IMethodSymbol
+
+            Dim vbReturnType = returnType.EnsureVbSymbolOrNothing(Of TypeSymbol)(NameOf(returnType))
+            Dim vbOperandType = returnType.EnsureVbSymbolOrNothing(Of NamedTypeSymbol)(NameOf(operandType))
+
+            Dim nameToCheck = If(SynthesizedIntrinsicOperatorSymbol.IsCheckedUnaryOperator(name), WellKnownMemberNames.UnaryNegationOperatorName, name)
```

Comment:
>SynthesizedIntrinsicOperatorSymbol.IsCheckedUnaryOperator(name) [](http://example.com/codeflow?start=33&length=63)

It is probably better to check for ```CheckedUnaryNegationOperatorName``` explicitly here rather than delegating to some other helper with general enough name to question the logic that follows.